packages/zod/src/index.ts

```diff
@@ -156,7 +156,7 @@ export const generateZodValidationSchemaDefinition = (
   const type = resolveZodType(schema);
   const required = rules?.required ?? false;
   const nullable =
-    schema.nullable ??
+    ('nullable' in schema && schema.nullable) ||
```

Comment:
this change was needed because `schema.nullable` does not exist in `SchemaObject31`